### ISSUE TITLE
PutMarkLogic processor drops some permissions

### ISSUE THREAD

**Issue by dmcassel** [marklogic/nifi#64]
The permissions property is a comma-delimited sequence of (role, capability) pairings. However, if you try to grant two capabilities to the same role, the first one gets dropped. Example:

    role1,update,role1,read

The result is that role1 will have read, but not update permissions on documents.

**Comment by rjrudin** [comment]
This should be a simple fix - the offending line is:

    metadata.withPermission(role, DocumentMetadataHandle.Capability.getValueOf(capability));

Which should instead do something like this:

```
if (permissions.containsKey(role)) {
  permissions.get(role).add(c);
} else {
  permissions.add(role, c);
}
```

Would be nice if "withPermission" worked that way too (which seems expected). 

@dmcassel Note there's a new marklogic-nifi-incubator project where this can probably be addressed more quickly, though in a non-supported release.

**Comment by rjrudin** [comment]
Addressed in incubator, which was merged into 1.9.1.5
